src/test/java/homework_4/custom_file_reader/CustomFileReaderTest.java

```diff
@@ -0,0 +1,41 @@
+package homework_4.custom_file_reader;
+
+import base.UnitBase;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import static homework_4.custom_file_reader.CustomFileReader.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+class CustomFileReaderTest extends UnitBase {
+
+    @Test
+    void TestRun1() throws IOException {
+        run1();
+        assertFalse(getOutput().contains("."));
+        assertFalse(getOutput().contains(","));
```

Comment:
Probably, what should you really test, is edge cases: what happens if file is empty? Or doesn't contain any dots or commas? Also, since you have setter for a file path in your CustomFileReader - what if we set a path of a non-existent file? 
And all of this test cases can be applied to all of the reading methods.

Reply:
"реализовать приложение, которое считывает данные из файла, и печатает в консоль, но уже без запятых и точек."
If there are no commas or dots in the file, or the file is empty, then these tests will pass perfectly.
The output should not contain dots and commas - this is the condition of the task. This is verified by tests.

"Файл - тот, что вы даете. Другого файла не ожидается. Все проще) "
Thanks for the comment, removed the setter.

Reply:
Add test - "removed only commas and dots" is a good idea

Reply:
_> If there are no commas or dots in the file, or the file is empty, then these tests will pass perfectly._ 

But what if you change your run method implementation in future? Test may fail in one situation and pass in others and that is not what is expected from tests. I mean:
1. The purpose of tests is to be able to add new code, or remove, or refactor, or whatever with being assure that you're not impacting the code you weren't plan to impact, that other app features work correctly. 
2. Tests need to be atomic, which means that one test check one test case and if it fails, it's crystal clear what is wrong. If your current tests fail, you certainly will understand which method is not working as expected (run1 or run2 etc), but you will need to test your app manually to get the concrete cause of test fail. Which will cost you some time. (test passed with ok and ordinary and expected file and test passed with empty file - these are two different test cases, because they have different data given).

I know, that hw4 wasn't about testing, nor about changing app in future (which, by the way, you never know), but your answer made me feel, that you're not properly understand testing, which might be important sometime in your career. You could search for some reliable sources about testing, if you want, it's quite good to know.
Also, whatever the task is, testing edge cases is must-have, in my opinion. 

Whatever, it's just the thoughts:)

Reply:
> "Файл - тот, что вы даете. Другого файла не ожидается. Все проще) " 

File - file name - could be the same, contents might differ